### DISCUSSION TITLE
Table Block Buttons do not update the state for the clicked row id

### DISCUSSION THREAD

**Discussion by thinkbig1979** [Budibase/budibase#13597] [Help] [answered]
**Hosting**
- Self
  - Method: Docker Compose
  - Budibase Version: 2.24.0.
- Cloud

**Describe the bug**
I have added buttons to my table that enable me to open a side panel to edit a related row in a different table.
If I click a cell in the row of the table first and then on the button, the correct related row is populated into the form block in the side panel. If I do not first select a row, but click on a button in a certain row, the fields are populated with the record matching the first entry in the table. In other words, the buttons on the rows do not update the {{ Clicked row._id }} state as you would imagine they would. I would expect that the button is associated with the row, and would update the {{ Clicked row._id }} the same as clicking on the row itself would.

It also highlights another item that I feel needs addressing. It seems that on page load, the {{ Clicked row._id }} state is popolated with the value from the first row of the table rather than being null. I would rather get an error saying "No row ID defined" than have my users accidentally edit the wrong row or create relationships between things they shouldn't be by mistake.

**Comment by aptkingston** [comment]
Hey @thinkbig1979. I think we'll need more details here - this is a very common flow that does work properly. Could you screenshot your bindings, because if you're using a button to open the side panel then you do not have a `{{ Clicked row._id }}` binding that you mention. That binding only exists if you're running actions when clicking a row.

When using buttons like you say, the binding would look like `{{ Table block.Foo._id }}`.

What you describe is almost exactly what we have by default in our "table with side panel" auto-screen, except we use row click actions rather than a button.

> It seems that on page load, the {{ Clicked row._id }} state is popolated with the value from the first row of the table rather than being null.

I'm not sure what you mean here. State values (as in `{{ State.x }}` bindings) will not have any value until you set them to something. If you've checked "Persist this value" then it might contain a value - it will attempt to load whatever value was last persisted. `{{ Clicked row._id }}` is a binding that only exists when running actions by clicking on a row, so does not exist until you click on a row.

**Comment by thinkbig1979** [comment, marked as answer]
Hi @aptkingston , I must apologize, I was totally convinced that I had set this up as you outlined above and double checked all my bindings, however, in the process of screenshotting the bindings I spotted my own error. 😳
I fixed the error and now it works as I had intended.
